database/boltdb.go

```diff
@@ -203,28 +146,50 @@ func (db *dbProvider) DbDelete(table string, key string) (err error) {
 	return err
 }
 
-func (db *dbProvider) DbGet(table string, key string) (value interface{}, err error) {
+func (db *dbProvider) DbGet(table string, key string, dbTable DbTable) (interface{}, error) {
 
-	err = db.DB.View(func(tx *bolt.Tx) error {
+	elem := db.Cache[table][key]
```

Comment:
Do we allow people to store nil values in the db?  As far as I can tell this is not disallowed by DbAdd, which means this code isn't correct as it cannot distinguish between the case where a key is actually missing from the cache and a key is present in the cache but the associated object is nil.  To actually see whether something is stored in the cache or not use.

elem, present := db.Cache[table][key]
if present {
   return elem, nil
}